lib/blather/stanza/x.rb

```diff
@@ -39,7 +39,7 @@ def self.new(type = nil, fields = [])
     # @param [Blather::Stanza] parent the parent node to search under
     # @return [Blather::Stanza::X]
     def self.find_or_create(parent)
-      if found_x = parent.find_first('//ns:x', :ns => self.registered_ns)
+      if found_x = parent.find_first('ns:x', :ns => self.registered_ns)
```

Comment:
It's actually fewer places, not more, before the change:

```ruby
x = Blather::Stanza::Iq::Command.new
x.form.fields = [{var: :a}]
p Blather::Stanza::X.find_or_create(x)
```

```xml
<x xmlns="jabber:x:data">
  <field var="a"/>
</x>
```

after:

```xml
<x xmlns="jabber:x:data"/>
```

This change makes it look only for direct children of the parent, whereas before it would recurse into all children looking for a matching element.

Reply:
The main reason was consistency, in the example above the fact that `Blather::X::Stanza.find_or_create(iq)` would return the form if present might lead one to actually use that code, but then when it's not present it gets created in the wrong place.  This kind of thing bit us several times.